internal/worker/bootstrap/service.go

```diff
@@ -32,6 +34,14 @@ type PasswordService interface {
 	SetUnitPassword(ctx context.Context, unitName unit.Name, password string) error
 }
 
+// AgentBinaryStore is responsible for persisting agent binary's into a long
+// term store for later retrival.
+type AgentBinaryStore interface {
+	// AddWithSHA256 adds a new agent binary to the object store and saves its
+	// metadata to the database.
+	AddWithSHA256(context.Context, io.Reader, coreagentbinary.Version, int64, string) error
```

Comment:
Did we want to name the params to make it explicitly clear what in64 and string are. You can reason that they are size and sha256 but naming is clearer.

Reply:
Yeah this is a good suggestion.